src/ethereum/byzantium/spec.py

```diff
@@ -185,6 +186,8 @@ def validate_header(header: Header, parent_header: Header) -> None:
     parent_header :
         Parent Header of the header to check for correctness
     """
+    ensure(header.timestamp > parent_header.timestamp, InvalidBlock)
```

Comment:
Why'd this move up here?

Reply:
Interesting. Should this check be moved into `calculate_block_difficulty` then?

Alternatively, would it make sense to move all the other checks (besides difficulty) up as well?

I worry that if we split the block of checks into several, it'll be less obvious.